src/osd/scrubber/pg_scrubber.cc

```diff
@@ -2426,45 +2528,187 @@ ostream& PgScrubber::show(ostream& out) const
   return out << " [ " << m_pg_id << ": " << m_flags << " ] ";
 }
 
-int PgScrubber::asok_debug(std::string_view cmd,
-			   std::string param,
-			   Formatter* f,
-			   stringstream& ss)
+
+int PgScrubber::asok_debug(
+    Formatter* f,
+    std::string_view prefix,
+    std::string_view cmd,
+    std::string_view param,
+    std::string_view arg1)
 {
-  dout(10) << __func__ << " cmd: " << cmd << " param: " << param << dendl;
+  dout(10) << fmt::format(
+		  "asok_debug: pr={} cmd={}, param={}, val={}", prefix, cmd,
+		  param, arg1)
+	   << dendl;
 
-  if (cmd == "block") {
-    // 'm_debug_blockrange' causes the next 'select_range' to report a blocked
-    // object
-    m_debug_blockrange = 10;  // >1, so that will trigger fast state reports
+  auto prfx_hash = cx_hash(prefix);
+  auto cmd_hash = cx_hash(cmd);
+  auto param_hash = cx_hash(param);
+
+  asok_op_cmd_t op_cmd = asok_op_cmd_t::unspecified;
+
+  // "normalize" the command parts
+  switch (prfx_hash) {
+    case cx_hash("scrubdebug_set"):
+      op_cmd = asok_op_cmd_t::set;
+      break;
+    case cx_hash("scrubdebug_clr"):
+      op_cmd = asok_op_cmd_t::clear;
+      break;
+    default:
+      break;
+  }
+
+  for (auto& e : PgScrubber::asok_dispatch_tbl) {
+    if (e.op_cmd != op_cmd) {
+      continue;
+    }
+    if (e.cmd_hash != cmd_hash) {
```

Comment:
Comparing just hashes is not enough, I'm afraid. It's enough to end iteration early but not to call the handler.